aws_lambda_powertools/utilities/data_classes/kafka_event.py

```diff
@@ -0,0 +1,122 @@
+        return self["topic"]
+
+    @property
+    def partition(self) -> str:
+        """The Kafka record parition."""
+        return self["partition"]
+
+    @property
+    def offset(self) -> str:
+        """The Kafka record offset."""
+        return self["offset"]
+
+    @property
+    def timestamp(self) -> int:
+        """The Kafka record timestamp."""
+        return self["timestamp"]
+
+    @property
+    def timestamp_type(self) -> str:
+        """The Kafka record timestamp type."""
+        return self["timestampType"]
+
+    @property
+    def key(self) -> str:
+        """The raw (base64 encoded) Kafka record key."""
+        return self["key"]
+
+    @property
+    def decoded_key(self) -> bytes:
+        """Decode the base64 encoded key as bytes."""
+        return base64.b64decode(self.key)
+
+    @property
+    def value(self) -> str:
+        """The raw (base64 encoded) Kafka record value."""
+        return self["value"]
+
+    @property
+    def decoded_value(self) -> bytes:
+        """Decodes the base64 encoded value as bytes."""
+        return base64.b64decode(self.value)
+
+    @property
+    def json_value(self) -> Any:
+        """Decodes the text encoded data as JSON."""
+        if self._json_data is None:
+            self._json_data = json.loads(self.decoded_value.decode("utf-8"))
+        return self._json_data
+
+    @property
+    def headers(self) -> List[Dict[str, List[int]]]:
+        """The raw Kafka record headers."""
+        return self["headers"]
+
+    @property
+    def decoded_headers(self) -> Dict[str, bytes]:
+        """Decodes the headers as a single dictionary."""
+        return {k: bytes(v) for chunk in self.headers for k, v in chunk.items()}
+
+    def get_header_value(
+        self, name: str, default_value: Optional[Any] = None, case_sensitive: bool = True
+    ) -> Optional[bytes]:
+        """Get a decoded header value by name."""
+        if case_sensitive:
+            return self.decoded_headers.get(name, default_value)
+        name_lower = name.lower()
+
+        return next(
+            # Iterate over the dict and do a case-insensitive key comparison
+            (value for key, value in self.decoded_headers.items() if key.lower() == name_lower),
+            # Default value is returned if no matches was found
+            default_value,
+        )
+
+
+class KafkaEvent(DictWrapper):
+    """Self-managed Apache Kafka event trigger
+    Documentation:
+    --------------
+    - https://docs.aws.amazon.com/lambda/latest/dg/with-kafka.html
+    """
+
+    @property
+    def event_source(self) -> str:
+        """The AWS service from which the Kafka event record originated."""
+        return self["eventSource"]
+
+    @property
+    def event_source_arn(self) -> str:
+        """The AWS service ARN from which the Kafka event record originated."""
+        return self["eventSourceArn"]
+
+    @property
+    def bootstrap_servers(self) -> Optional[List[str]]:
```

Comment:
Hi @lyoung-confluent! I don't think we need to add this "if" here. Even the AWS documentation [here](https://docs.aws.amazon.com/lambda/latest/dg/with-msk.html) doesn't mention the bootstrapServers field, I created an MSK cluster and added it as an event source of a Lambda and I was able to confirm that the bootstrapServers field is also present there.
![image](https://user-images.githubusercontent.com/4295173/188456575-7f97cbb8-cd66-4905-b730-b055ae034c25.png)

btw, I've notified the documentation team to include it.

Reply:
Great, I've updated it to assume the field is always present and just return `List[str]` instead. 

Do you think it makes sense though for this property to return a list or should it return the raw string and we offer a `decoded_bootstrap_servers` that does the split similar to other `decoded_` properties?

Reply:
OK I see! Great idea, so in this case we can have the raw string and another property with a decoded value like List.

Please, do this.